### DISCUSSION TITLE
Is there any way that I can scroll entire page in Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#17445] [Q&A]
What I actually want to do is to achieve the full-page flipping effect in iOS as shown in the following GIF.
![2](https://github.com/user-attachments/assets/9842c622-f194-4db6-b705-5b2e1e73e79e)

Once scrolled, the entire page is scrolled instead of just a certain distance.

I just thought of placing a thumb on the top of the page, and then turning the page by judging the drag of the thumb. But I think this way is very troublesome, and the existence of the thumb will also affect the click of some controls in the page.

I'm not sure if Avalonia can achieve this natively. Can you tell me a better way to do it?

**Comment by thevortexcloud** [comment]
This just looks like a carousel to me. Not sure how well it works with touch though.

https://docs.avaloniaui.net/docs/reference/controls/carousel

**Reply by kyuranger** [reply]
I'm not sure how it works in Linux DRM, but looking at the Avalonia tutorial it seems to require a separate button to turn the pages. Maybe I can test it tomorrow at work.

**Reply by thevortexcloud** [reply]
You can _probably_ hook a gesture up to it.

https://docs.avaloniaui.net/docs/reference/gestures/scrollgesturerecognizer

It would probably require some fiddling though to make it actually follow the position of a finger like your video shows though. But I also don't have much experience with gestures.
